### PR TITLE
Restore users, before data

### PR DESCRIPTION
**Pull request by UserNotFound**
Views that have a `SECURITY DEFINER` set can only be restored from the `mysqldump` backup if the users have already been created.

This is achieved by dumping the `mysql` database by itself, and restoring it (ie mysql.user) first, followed by a `FLUSH PRIVILEGES`, before the views are restored.

### CODE REVIEW

**Comment by krallin** [comment]
Can we add some tests or at least refer to documentation explaining why this works? This is a little obscure as-is.

File: bin/run-database.sh [hunk 1]
```diff
@@ -276,9 +277,14 @@ elif [[ "$1" == "--initialize-from" ]]; then
 
   # shellcheck disable=SC2154
   MYSQL_PWD="$password" mysqldump --host "$host" --port "${port:-$DEFAULT_PORT}" --user "$MYSQL_REPLICATION_ROOT" --ssl-mode=REQUIRED --ssl-cipher="${SSL_CIPHERS}" \
-    --all-databases --master-data \
+    mysql --flush-privileges \
     > "${MASTER_DUMPFILE}"
```
Comment (krallin) [inline bin/run-database.sh]: The naming is confusing here (MASTER_DUMPFILE as opposed to DB_DUMPFILE).

Can we rename those "PRIVILEGES_DUMPFILE" and "DATA_DUMPFILE" or something?

**Comment by UserNotFound** [comment]
Test added : the specific issue encountered is when a view of a view is restored.

If you omit `mysql mysql < "${PRIVILEGES_DUMPFILE}"` from run-database.sh (ie remove the patch this PR provides), the replication test will fail with the error `The user specified as a definer ('root'@'%') does not exist`.

**Comment by UserNotFound** [comment]
@krallin I needed to update to the latest revisions here to get this in a merge-able state, so I guess this should get a second review

https://dev.mysql.com/doc/relnotes/mysql/8.0/en/news-8-0-13.html
https://dev.mysql.com/doc/relnotes/mysql/5.7/en/news-5-7-24.html
https://dev.mysql.com/doc/relnotes/mysql/5.6/en/news-5-6-42.html

**Comment by krallin** [comment]
@UserNotFound looks good to go :+1:

**Pull request merged**